bvm/ballerina-core/src/main/java/org/ballerinalang/model/types/BArrayType.java

```diff
@@ -74,20 +69,20 @@ public BType getElementType() {
             int tag = elementType.getTag();
             switch (tag) {
```

Comment:
i think we can remove this switch case now, and simply return `new BValueArray(elementType, size);`

Reply:
We cannot fully remove the switch case because there is a special case for DecimalArray and also Ref Array. But we could refine the switch a bit.